File: Wonderful Sticks.js
function solve(input) {
    const lines = input.trim().split('\n');
    let t = parseInt(lines[0]);
    let lineIndex = 1;

    for (let _ = 0; _ < t; _++) {
        let n = parseInt(lines[lineIndex++]);
        let s = lines[lineIndex++].trim();

        let count_L = (s.match(/</g) || []).length;
        let count_R = (s.match(/>/g) || []).length;

        let small = [];
        for (let i = 1; i <= count_L; i++) small.push(i);

        let large = [];
        for (let i = n - count_R + 1; i <= n; i++) large.push(i);

        let full_set = new Set();
        for (let i = 1; i <= n; i++) full_set.add(i);
        let mid_vals = new Set(full_set);
        small.forEach(val => mid_vals.delete(val));
        large.forEach(val => mid_vals.delete(val));

        let mid = [...mid_vals][0];

        let demands = [...s].map(ch => (ch === '<' ? 'S' : 'L'));

        let ans = new Array(n);
        ans[0] = mid;
        let pos = 1;

        let i = 0;
        while (i < demands.length) {
            let tag = demands[i];
            let j = i + 1;
            while (j < demands.length && demands[j] === tag) j++;
            let length = j - i;

            if (tag === 'S') {
                let block = small.slice(-length).sort((a, b) => b - a);
                small.length -= length;
                for (let x of block) ans[pos++] = x;
            } else {
                let block = large.slice(0, length).sort((a, b) => a - b);
                large = large.slice(length);
                for (let x of block) ans[pos++] = x;
            }
            i = j;
        }

        console.log(ans.join(' '));
    }
}

// Read from stdin
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8');
solve(input);

File: Wonderful Sticks.py
import sys

def main():
    input = sys.stdin.readline
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input().strip()
        # Total '<' and '>' in s
        count_L = s.count('<')
        count_R = s.count('>')
        # RegionSmall for '<' picks, RegionLarge for '>' picks, RegionMid for a[1]
        small = list(range(1, count_L + 1))
        large = list(range(n - count_R + 1, n + 1))
        full_set = set(range(1, n + 1))
        mid_vals = full_set - set(small) - set(large)
        # There should be exactly one mid value
        mid = mid_vals.pop()

        # Build demands for a[2..n]: 'S' for small, 'L' for large
        demands = ['S' if ch == '<' else 'L' for ch in s]

        ans = [None] * n
        ans[0] = mid
        pos = 1  # Next position to fill in ans (0-based)

        i = 0
        while i < len(demands):
            tag = demands[i]
            j = i + 1
            while j < len(demands) and demands[j] == tag:
                j += 1
            length = j - i

            if tag == 'S':
                # Pick 'length' largest from small, assign in descending order
                block = small[-length:]
                small = small[:-length]
                block.sort(reverse=True)
                for x in block:
                    ans[pos] = x
                    pos += 1
            else:
                # tag == 'L'
                # Pick 'length' smallest from large, assign in ascending order
                block = large[:length]
                large = large[length:]
                block.sort()
                for x in block:
                    ans[pos] = x
                    pos += 1

            i = j

        print(' '.join(map(str, ans)))

if __name__ == "__main__":
    main()